app/store/charmstore-api.js

```diff
@@ -345,6 +345,43 @@ YUI.add('charmstore-api', function(Y) {
           this.apiPath, charmId, 'icon.svg'].join('/');
       }
       return path;
+    },
+
+    /**
+      Takes the bundle id and fetches the bundle YAML contents. Required for
+      deploying a bundle via the deployer.
+
+      @method getBundleYAML
+      @param {String} id Bundle id in apiv4 format.
+      @param {Function} successCallback The success callback.
+      @param {Function} failureCallback The failure callback.
+    */
+    getBundleYAML: function(id, successCallback, failureCallback) {
+      this.getEntity(
+          id, this._getBundleYAMLResponse.bind(
+              this, successCallback, failureCallback), failureCallback);
```

Comment:
I'm not sure this call is right, based on the def below. It looks like your passing in context first, the callbacks, and no bundle at all to the `.bind` call on `_getBundleYAMLResponse`. Am I misreading this?